src/compile/legend/encode.ts

```diff
@@ -18,7 +18,16 @@ import {getFirstDefined, keys} from '../../util';
 import {applyMarkConfig, timeFormatExpression} from '../common';
 import * as mixins from '../mark/mixins';
 import {UnitModel} from '../unit';
+import {ScaleChannel} from './../../channel';
 import {LegendComponent} from './component';
+import {defaultType} from './properties';
+
+function type(legendCmp: LegendComponent, model: UnitModel, channel: Channel) {
```

Comment:
channel here should be `ScaleChannel` otherwise it can't have a legend.

Reply:
Done